packages/react-shopify-app-bridge/spec/Provider.spec.tsx

```diff
@@ -210,103 +213,159 @@ describe("GadgetProvider", () => {
+    mockUrqlClient.executeMutation.pushResponse("ShopifyFetchOrInstallShop", {
+      data: {
+        shopifyConnection: {
+          fetchOrInstallShop: {
+            isAuthenticated: false,
+            redirectToOauth: false,
           },
         },
-        stale: false,
-        hasNext: false,
-      });
-
-      expect(container.outerHTML).toMatchInlineSnapshot(`"<div><span>hello world</span></div>"`);
-      expect(mockOpen).not.toHaveBeenCalled();
-      expect(mockNavigate).not.toHaveBeenCalled();
-
-      window.shopify.environment.embedded = false;
+      },
+      stale: false,
+      hasNext: false,
     });
 
-    test("can render an embedded app type that's already authenticated", async () => {
-      window.shopify.environment.embedded = true;
+    expect(container.outerHTML).toMatchInlineSnapshot(`"<div><span>hello world</span></div>"`);
+    expect(mockOpen).not.toHaveBeenCalled();
+    expect(mockNavigate).not.toHaveBeenCalled();
+  });
 
-      const { container } = render(
-        <Provider api={mockApiClient} shopifyApiKey={mockApiKey} type={AppType.Embedded}>
-          <span>hello world</span>
-        </Provider>
-      );
+  test("can render an embedded app type that's already authenticated", async () => {
+    window.shopify.environment.embedded = true;
 
-      await act(async () => {
-        resolveIdToken("mock-id-token");
-        await mockUrqlClient.executeMutation.waitForSubject("ShopifyFetchOrInstallShop");
-      });
+    const { container } = render(
+      <Provider api={mockApiClient} shopifyApiKey={mockApiKey} type={AppType.Embedded}>
+        <span>hello world</span>
+      </Provider>
+    );
 
-      mockUrqlClient.executeMutation.pushResponse("ShopifyFetchOrInstallShop", {
-        data: {
-          shopifyConnection: {
-            fetchOrInstallShop: {
-              isAuthenticated: true,
-              redirectToOauth: false,
-            },
+    await act(async () => {
+      resolveIdToken("mock-id-token");
+      await mockUrqlClient.executeMutation.waitForSubject("ShopifyFetchOrInstallShop");
+    });
+
+    mockUrqlClient.executeMutation.pushResponse("ShopifyFetchOrInstallShop", {
+      data: {
+        shopifyConnection: {
+          fetchOrInstallShop: {
+            isAuthenticated: true,
+            redirectToOauth: false,
           },
         },
-        stale: false,
-        hasNext: false,
-      });
+      },
+      stale: false,
+      hasNext: false,
+    });
 
-      expect(container.outerHTML).toMatchInlineSnapshot(`"<div><span>hello world</span></div>"`);
-      expect(mockNavigate).not.toHaveBeenCalled();
+    expect(container.outerHTML).toMatchInlineSnapshot(`"<div><span>hello world</span></div>"`);
+    expect(mockNavigate).not.toHaveBeenCalled();
 
-      window.shopify.environment.embedded = false;
+    window.shopify.environment.embedded = false;
+  });
+
+  test("can render an embedded app with an install state hint saying we're installed", async () => {
```

Comment:
should we add a test that covers the case where the mutation comes back with a result that contradicts the hint?

Reply:
Added in the other file where the results are easier to assert